x-pack/plugin/ilm/src/main/java/org/elasticsearch/xpack/ilm/IlmHealthIndicatorService.java

```diff
@@ -79,39 +170,179 @@ public HealthIndicatorResult calculate(boolean verbose, int maxAffectedResources
                 Collections.emptyList()
             );
         } else if (currentMode != OperationMode.RUNNING) {
-            List<HealthIndicatorImpact> impacts = Collections.singletonList(
-                new HealthIndicatorImpact(
-                    NAME,
-                    AUTOMATION_DISABLED_IMPACT_ID,
-                    3,
-                    "Automatic index lifecycle and data retention management is disabled. The performance and stability of the cluster "
-                        + "could be impacted.",
-                    List.of(ImpactArea.DEPLOYMENT_MANAGEMENT)
-                )
-            );
             return createIndicator(
                 YELLOW,
                 "Index Lifecycle Management is not running",
                 createDetails(verbose, ilmMetadata, currentMode),
-                impacts,
+                AUTOMATION_DISABLED_IMPACT,
                 List.of(ILM_NOT_RUNNING)
             );
         } else {
+            return calculateIndicator(verbose, ilmMetadata, currentMode, maxAffectedResourcesCount);
+        }
+    }
+
+    private HealthIndicatorDetails createDetails(boolean verbose, IndexLifecycleMetadata ilmMetadata, OperationMode currentMode) {
+        return createDetails(verbose, ilmMetadata, currentMode, List.of());
+    }
+
+    private HealthIndicatorResult calculateIndicator(
+        boolean verbose,
+        IndexLifecycleMetadata ilmMetadata,
+        OperationMode currentMode,
+        int maxAffectedResourcesCount
+    ) {
+        var stuckIndices = stuckIndicesFinder.find();
+
+        if (stuckIndices.isEmpty()) {
             return createIndicator(
                 GREEN,
                 "Index Lifecycle Management is running",
-                createDetails(verbose, ilmMetadata, currentMode),
+                createDetails(verbose, ilmMetadata, currentMode, stuckIndices),
                 Collections.emptyList(),
                 Collections.emptyList()
             );
         }
+
+        return createIndicator(
+            YELLOW,
+            (stuckIndices.size() > 1 ? "Some indices have" : "An index has") + " been stuck on the same action longer than expected.",
+            createDetails(verbose, ilmMetadata, currentMode, stuckIndices),
+            INDEX_STUCK_IMPACT,
```

Comment:
Nit: What about stagnating indices instead of stuck indices? Then, this would be `STAGNATING_INDEX_IMPACT`?